tests/test_inject_ib_ids.py

```diff
@@ -0,0 +1,158 @@
+  </div>
+  <div class="os-raise-ib-cta-prompt">
+    Here is a prompt for a CTA button
+  </div>
+</div>
+<p>Here is some intermediary text</p>
+<div class="os-raise-ib-content" data-wait-for-event="eventnameX"
+data-schema-version="1.0">
+  Here is some conditional content that waits on the CTA button.
+</div>
+<div class="os-raise-ib-input" data-button-text="Submit"
+data-schema-version="1.0">
+  <div class="os-raise-ib-input-content">
+    This is the question for an IB input block (5+6)
+  </div>
+  <div class="os-raise-ib-input-prompt">
+    Here is the prompt for the submission of the IB Input Block
+  </div>
+  <div class="os-raise-ib-input-ack">
+    This is the ack for the input block - Answer was 12
+  </div>
+</div>
+""".strip()
+
+    page_2_html = """
+<h1>This is the second page</h1>
+<div class="os-raise-ib-pset" data-fire-success-event="eventnameX"
+data-fire-learning-opportunity-event="eventnameY" data-retry-limit="3"
+data-button-text="Check" data-schema-version="1.0">
+  <div class="os-raise-ib-pset-problem" data-problem-type="input"
+  data-solution="42" data-problem-comparator="integer">
+    <div class="os-raise-ib-pset-problem-content">
+      <p>Here is content for an input problem</p>
+    </div>
+  </div>
+  <div class="os-raise-ib-pset-problem" data-problem-type="dropdown"
+  data-solution="red" data-solution-options='["red", "blue", "green"]'>
+    <div class="os-raise-ib-pset-problem-content">
+      <p>Here is content for a dropdown problem</p>
+    </div>
+  </div>
+  <div class="os-raise-ib-pset-problem" data-problem-type="multiplechoice"
+  data-solution="red" data-solution-options='["red", "blue", "green"]'>
+    <div class="os-raise-ib-pset-problem-content">
+      <p>Here is content for a multiple choice problem</p>
+    </div>
+  </div>
+  <div class="os-raise-ib-pset-problem" data-problem-type="multiselect"
+  data-solution='["red", "blue"]'
+  data-solution-options='["red", "blue", "green"]'>
+    <div class="os-raise-ib-pset-problem-content">
+      <p>Here is content for a multiselect problem</p>
+    </div>
+  </div>
+  <div class="os-raise-ib-pset-correct-response">
+    <p>If you got an answer right you'll see this response</p>
+  </div>
+  <div class="os-raise-ib-pset-encourage-response">
+    <p>If you got an answer wrong you'll see this prompt</p>
+  </div>
+</div>
+""".strip()
+
+    os.mkdir(tmp_path / "html")
+    html_dir = f"{tmp_path}/html"
+    fp1 = Path(html_dir) / "page_1.html"
+    fp2 = Path(html_dir) / "page_2.html"
+    fp1.write_text(page_1_html)
+    fp2.write_text(page_2_html)
+
+    mocker.patch(
+        "mbtools.models.uuid4",
+        lambda: autogenerated_user_uuid
+    )
+
+    mocker.patch(
+        "sys.argv",
+        ["", html_dir]
+    )
+
+    inject_ib_ids.main()
+
+    with open(fp1, 'r') as f:
+        assert (f.read().count("data-content-id") == 1)
+
+    with open(fp2, 'r') as f:
+        assert (f.read().count("data-content-id") == 5)
```

Comment:
It would be nice to have these tests actually assert on the generated output vs characteristics of the output like we do in [these tests](https://github.com/openstax/raise-mbtools/blob/main/tests/test_fix_html.py). The reason being in the past we've been bitten by unexpected serialization issues, and these assertions won't catch stuff like that.

Reply:
Ok, so would you prefer the entire correct result sit in the test case like a string? I was playing with that earlier and it looked like it'd get long and messy. Any better way?

Reply:
:man_shrugging: - The two options that come immediately to mind are in-line strings like we do in the `fix-html` tests or static input / expect files in a `tests/data` like we have in `raise-enclave`. The latter might be better if you think the in-lines are too long (even if you reduce them so they're still representative but don't have unnecessary fluff). If so, we may want to structure the files in `tests/data` in a way that makes it clear where files are meant for specific test cases.

Reply:
Oh, just as a note, I'm fairly certain `bs4` is going to create white space changes which is why @P-Gill97 probably setup his fixtures as a tuple of strings (and a reason for why we ended up running `js-beautify` in [these steps](https://docs.google.com/document/d/1yP_BV827GtOP-rscESF78BaVWZ1Bit3DXKB6lubMrQg/edit#bookmark=id.2y3seaitob9p)).